src/sentry/incidents/logic.py

```diff
@@ -788,6 +807,15 @@ def delete_alert_rule(alert_rule, user=None):
     if alert_rule.status == AlertRuleStatus.SNAPSHOT.value:
         raise AlreadyDeletedError()
 
+    if user:
+        create_audit_entry_from_user(
+            user,
+            organization_id=alert_rule.organization_id,
+            target_object=alert_rule.id,
+            data=alert_rule.get_audit_log_data(),
+            event=AuditLogEntryEvent.ALERT_RULE_REMOVE,
+        )
```

Comment:
It probably makes sense to include this inside the transaction, so that if anything goes wrong we don't end up with an orphaned record here.